builtin/credential/approle/backend_test.go

```diff
@@ -26,6 +27,39 @@ func createBackendWithStorage(t *testing.T) (*backend, logical.Storage) {
 	return b, config.StorageView
 }
 
+func TestAppRole_RoleServiceToBatchNumUses(t *testing.T) {
+	b, s := createBackendWithStorage(t)
+
+	requestFunc := func(operation logical.Operation, data map[string]interface{}) {
+		resp, err := b.HandleRequest(context.Background(), &logical.Request{
+			Path:      "role/testrole",
+			Operation: operation,
+			Storage:   s,
+			Data:      data,
+		})
+		if err != nil || (resp != nil && resp.IsError()) {
+			t.Fatalf("bad: err: %#v\nresp: %#v", err, resp)
+		}
+	}
+
+	data := map[string]interface{}{
+		"bind_secret_id":        false,
+		"secret_id_bound_cidrs": "10.90.28.0/22",
+		"secret_id_num_uses":    0,
+		"secret_id_ttl":         "10m",
+		"token_policies":        "policy",
+		"token_ttl":             "5m",
+		"token_max_ttl":         "10m",
+		"token_num_uses":        2,
+		"token_type":            "default",
+	}
+	requestFunc(logical.CreateOperation, data)
+
+	data["token_num_uses"] = 0
+	data["token_type"] = "batch"
+	requestFunc(logical.UpdateOperation, data)
```

Comment:
Would it make sense to try using the token after making the change?

Reply:
I am not sure I understand. This is when updating a role that is used to issue a token. Do you mean that that test should issue a token and try to use it?

Reply:
The tests stops here because without the fix, the update step used to fail and now it doesn't.

Reply:
Maybe?  I just figure if this wasn't possible before, the resulting role might not yield usable tokens due to other latent bugs, so it would be nice to validate that they work.

Reply:
Sure. I'll add those steps.

Reply:
This is a backend level unit test. So I couldn't ensure that the token is usable. However the test now does ensure that a valid Auth block is returned from the engine. The token creation and usability should not be affected in my opinion. Pushed up the changes.